convert ethers's representation of infinite keys to UNLIMITED_KEYS_COUNT

cellog opened the pull request:
# Description

One of the details of Unlock that will cause issues when upgrading to use ethers is that infinite keys is represented internally slightly differently.

This PR provides a thin layer of abstraction that will allow us to clearly communicate to ethers the value of infinity without hard-coding it in the apps.

When installing the ethers version of the app, we will need to update walletMiddleware listening to `CREATE_LOCK` and then convert our app representation of infinity (`-1`) to `unlock-js`'s representation (`INFINITE_KEYS`). Note that web3 used a totally different system to represent the uint256 value we use for infinity, so this breaks creation of infinite keys

# Issues

<!-- This PR should fix or reference at least one existing issue ID. Add or delete as appropriate. -->

Refs #2782 

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [X] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

<!--
PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)
-->

julien51 commented:
It'd much prefer if we did not have to change the logic of the apps that implement this.
Can we rather make sure that the value is converted to what we used to have with web3js?

In the same way that we mapped `totalSupply` to `outstandingKeys`?

cellog commented:
Yeah I think that's best, I'll refactor this to do that

cellog commented:
ok @julien51 I refactored to convert from `UNLIMITED_KEYS_COUNT` to `ETHERS_MAX_UINT256` inside `createLock`. I also added a new test to verify that the conversion happens but we still emit the correct value of `UNLIMITED_KEYS_COUNT` in `'lock.updated'`.

I also amended the `getLock` tests to ensure that when we have a lock with unlimited keys, we return `UNLIMITED_KEYS_COUNT` and not `ETHERS_MAX_UINT256`
